### PR TITLE
roachtest: allow opting out of post validation(s)

### PR DESCRIPTION
**Pull request by herkolategan**
A few post validations are currently performed after a test completes. In most cases it applies to all tests, but there are some tests that are incompatible with particular post validations. This change adds the ability for a test to specify that it wants to opt-out of certain validations.

Refs: https://github.com/cockroachdb/cockroach/issues/97912
Epic: None

### CODE REVIEW

File: pkg/cmd/roachtest/registry/test_spec.go [hunk 1]
```diff
@@ -84,10 +84,28 @@ type TestSpec struct {
...
+
 	// Run is the test function.
 	Run func(ctx context.Context, t test.Test, c cluster.Cluster)
 }
 
+// PostValidation is a type of post-validation that runs after a test completes.
+type PostValidation int
+
+const (
+	// PostValidationReplicaDivergence is a post-validation that checks for
```
Comment (aliher1911) [inline pkg/cmd/roachtest/registry/test_spec.go:101]: nit: those comments look unhelpful. they just expand the name into sentence. maybe we can just refer to underlying sql query that it performs? e.g. is a post validation that fails test if `crdb_internal.check_consistency` finds replicas in inconsistent state. same for `PostValidationInvalidDescriptors` and `crdb_internal. invalid_objects`.

**Review by aliher1911 — APPROVED** [review]
Some nits about comments.
💯

**Review by herkolategan — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/98848)* status: :shipit: complete! 0 of 0 LGTMs obtained (waiting on @aliher1911, @renatolabs, and @srosenberg)

___
*[`pkg/cmd/roachtest/registry/test_spec.go` line 87 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/98848#-NQjGUisZp93OzoKfq79-r1-87:-NQyaFgP2de_gUFtphT0:b-ofbzf0) ([raw file](https://github.com/cockroachdb/cockroach/blob/8127d0c38e732a3c563c561093405d11a4493a20/pkg/cmd/roachtest/registry/test_spec.go#L87)):*
<details><summary><i>Previously, aliher1911 (Oleg) wrote…</i></summary><blockquote>

nit: bit-set maybe to make it more clear that you need to | all skipped validations.
</blockquote></details>

good idea, that should make it clearer.
___
*[`pkg/cmd/roachtest/registry/test_spec.go` line 101 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/98848#-NQjGUisZp93OzoKfq79-r1-101:-NQyaPFKB_68W0XUxRYM:b-ekgjfo) ([raw file](https://github.com/cockroachdb/cockroach/blob/8127d0c38e732a3c563c561093405d11a4493a20/pkg/cmd/roachtest/registry/test_spec.go#L101)):*
<details><summary><i>Previously, aliher1911 (Oleg) wrote…</i></summary><blockquote>

nit: those comments look unhelpful. they just expand the name into sentence. maybe we can just refer to underlying sql query that it performs? e.g. is a post validation that fails test if `crdb_internal.check_consistency` finds replicas in inconsistent state. same for `PostValidationInvalidDescriptors` and `crdb_internal. invalid_objects`.
</blockquote></details>

Fair point, those comments are pretty bland. There are comments on the functions that do the validations explaining how they work, but might be worth-while duplicating it or moving it here for easier reference.

**Comment by aliher1911** [comment]
We should think about backporting this to release as we don't want 23.1 branch to flake going further.

**Comment by herkolategan** [comment]
bors r=aliher1911

**Comment by herkolategan** [comment]
bors retry

**Pull request merged**